test/indexes-check.test.js

```diff
@@ -0,0 +1,147 @@
+// we were seeing weird behaviour in ssb-backlinks
+// this test is designed to pin that down
+
+const test = require('tape')
+const pull = require('pull-stream')
+
+const { Server } = require('./helpers')
+
+const PROFILE = 'profile/person'
+const LINK = 'link/feed-profile'
+
+function createRecords (server, t, cb) {
+  // this function creates a group and published a handful of messages to that group
+  // it calls back with the groupId and the list of published messages
+
+  /* state */
+  var published = [] // contents of messages we published
+  var lastMsgId
+
+  server.tribes.create({}, (err, { groupId } = {}) => {
+    t.error(err, 'creates group')
+
+    var recps = [groupId]
+    // recps = null // passes all tests
```

Comment:
NOTE: uncomment this line and all the tests pass ):

Reply:
Also just checked with `ssb-private1` style encryption, same problem there.
So it's something to do with the box/ unbox pipeline ...